Remove unneeded IValue copy in unpickler.

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #22953 Add initial support for serializing classes
* **#22883 Remove unneeded IValue copy in unpickler.**
* #22882 Cleanup some logic in pickler

Differential Revision: [D16270330](https://our.internmc.facebook.com/intern/diff/D16270330)

zdevito committed:
Remove unneeded IValue copy in unpickler.

driazati commented on torch/csrc/jit/pickler.cpp:
```diff
@@ -714,17 +714,10 @@ OpCode Unpickler::readInstruction() {
         case PicklerClass::BOOLLIST:
-          stack_.emplace_back(data->elements().at(0).toBoolList());
-          break;
+          stack_.emplace_back(data);
```
Shouldn't this unwrap it from the tuple so these don't get returned as 1-element tuples? i.e. `emplace_back(data->elements().at(0))`

zdevito committed:
Update on "Remove unneeded IValue copy in unpickler."

Remove unneeded IValue copy in unpickler.

gh-metadata: pytorch pytorch 22883 gh/zdevito/72/head
